### PR TITLE
9 raise error when model doesnt fit the task

### LINKED ISSUE #9: Raise error when model doesn't fit the task
If we run `evaluate -m bert-base-uncased -t ner` then we don't get any errors, but instead it attempts to evaluate the model and just stops.

Instead, it should give an informative error message.

### PR DESCRIPTION
**Pull request by AJDERS**
This PR fixes #9 

* Added new exception `InvalidArchitectureForTask`.
* Added comparison of `supertask` and the huggingface `config.architectures`, if they dont match throw `InvalidArchitectureForTask`.

### CODE REVIEW

File: src/aiai_eval/task.py [hunk 1]
```diff
@@ -544,8 +545,22 @@ def _load_pytorch_model(
...
+                # Check if model architecture fit the supertask of the provided task.
+                if not any(
+                    ["SequenceClassification" in arc for arc in config.architectures]
+                ):
+                    raise InvalidArchitectureForTask(
+                        architectures=config.architectures, supertask=supertask
+                    )
                 model_cls = AutoModelForSequenceClassification  # type: ignore
             else:
                 raise ValueError(f"The supertask `{supertask}` was not recognised.")
```
Comment (saattrupdan) [inline src/aiai_eval/task.py:566]: Maybe move this to a separate `_check_supertask` method?

File: src/aiai_eval/task.py [hunk 2]
```diff
@@ -602,6 +615,14 @@ def _load_pytorch_model(
 
         return dict(model=model, tokenizer=tokenizer)
 
+    def _check_supertask(
+        self, architectures: Sequence[str], supertask: str, search_str: str
+    ):
```
Comment (saattrupdan) [inline src/aiai_eval/task.py:620]: Missing docstring, including the `Raises:` block

**Review by saattrupdan — APPROVED** [review]
One small docstring left - feel free to merge with or without it 🙂

File: src/aiai_eval/task.py (hunk 2 above)
Comment (AJDERS) [reply src/aiai_eval/task.py:620]: I am way to forgetful...

**Review by saattrupdan — APPROVED** [review]
LGTM!

**Pull request merged**